### ISSUE TITLE
race condition

### ISSUE THREAD

**Issue by Mridul1911** [Ajaykumaryd/bookmyShow#1]
does your program also handles the condition when two users are trying to book same seat, ie race condition?

**Comment by Ajaykumaryd** [comment]
No I don't remember as such thing  but we can handle it just go through
deadlock conditions, we can do it.

**Comment by Mridul1911** [comment]
actually i want to learn springboot and i saw your repo, i found it impressive but usually in such reservation systems race condition must be handled, to avoid multiple booking of same seat by many users. thats why.
But i will understand your code and will see how can i handle this situation

**Comment by Ajaykumaryd** [comment]
Yes exactly but for a fresher it is good to have this much knowledge
handling race conditions and multi threading situations is something which
requires ample knowledge when I made this I was not aware of many things
but know when I am working in industry I have solved these multi threading
issues on live projects not very complex but that could be solved.Its
something that is not given to a freshie even developers having more then
two years of experience don't have this much knowledge of multi
threading.So just go through the basics and learn the implementation.Rest
all you will be able to learn while working.

**Comment by Mridul1911** [comment]
ok thanks buddy

**Comment by parthdeokate26** [comment]
> actually i want to learn springboot and i saw your repo, i found it impressive but usually in such reservation systems race condition must be handled, to avoid multiple booking of same seat by many users. thats why. But i will understand your code and will see how can i handle this situation

Hey did anyone of you added a race condition were you able to add it ??
